### PR TITLE
Add lint check support for kotlin extensions 

### PR DESCRIPTION
**Pull request by Rajin9601**
**Description**:

Lint check should be able to check RxKotlin's subscribeBy.
Looking at the discussion in #304, I get that the library doesn't want to add support only for RxKotlin because it's not part of the api.

So this PR supports kotlin extensions with configuration in gradle.properties. 

For the concern that it would slow down lint-check, I couldn't check for big project but I think this implementation would not slow down lint check much for those who don't use the added configuration. 

**Related issue(s)**: #304

### CODE REVIEW

File: docs/lint-check.md [hunk 1]
```diff
@@ -61,6 +61,14 @@ autodispose.lenient=true
 
 The default value of this is `false`. 
 
+### Kotlin Extension
+
+By default, subscribe and subscribeWith methods are checked. To support other subscribe methods such as subscribeBy in RxKotlin, you can add your own subscribe extensions.
+In your **app-level** `gradle.properties` files, add kotlin extension functions in format of `{full package name for extension's scope}#{functionName}` and comma-separated-values like so:
+```groovy
```
Comment (ZacSweers) [inline docs/lint-check.md:68]:
```suggestion
In your **app-level** `gradle.properties` files, add kotlin extension functions in format of `{full package name for extension's scope}#{functionName}` and comma-separated-values like so:

```groovy
```

File: docs/lint-check.md [hunk 2]
```diff
@@ -61,6 +61,14 @@ autodispose.lenient=true
 
 The default value of this is `false`. 
 
+### Kotlin Extension
+
+By default, subscribe and subscribeWith methods are checked. To support other subscribe methods such as subscribeBy in RxKotlin, you can add your own subscribe extensions.
```
Comment (ZacSweers) [inline docs/lint-check.md:66]:
```suggestion
By default, `subscribe` and `subscribeWith` methods are checked. To support other subscribe methods such as `subscribeBy` in RxKotlin, you can add your own subscribe extensions.
```

**Review by ZacSweers — COMMENTED** [review]
Seems ok to me, apologies for the delay. @ShaishavGandhi what do you think?

**Review by ShaishavGandhi — APPROVED** [review]
LGTM! Let me take a look at the build issues and then we can merge it.

**Comment by Rajin9601** [comment]
I updated this branch with `uber:main` and passed the CI tests!
It seems like this PR can be merged now.

**Comment by ZacSweers** [comment]
Thanks!

**Pull request merged**